tests/helpers/process_group.py

```diff
@@ -0,0 +1,44 @@
+# Copyright The PyTorch Lightning team.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import os
+from contextlib import contextmanager
+
+import torch.distributed as dist
+
+from pytorch_lightning.plugins.environments.lightning_environment import find_free_network_port
+
+
+@contextmanager
+def single_process_pg():
```

Comment:
This could be a `pytest.fixture. See: https://github.com/PyTorchLightning/pytorch-lightning/blob/master/tests/conftest.py

Reply:
Very cool! Thanks for the suggestion!